rcl_action/src/rcl_action/action_client.c

```diff
@@ -649,6 +701,209 @@ rcl_action_client_wait_set_get_entities_ready(
   return RCL_RET_OK;
 }
 
+static char*
+to_uuid_string(const uint8_t * uuid, rcl_allocator_t allocator) {
+  char * uuid_str = rcutils_format_string(allocator, UUID_FMT, UUID_FMT_ARGS(uuid));
+  return uuid_str;
+}
+
+static
+rcl_ret_t set_content_filtered_topic(
+  const rcl_action_client_t * action_client)
+{
+  rcl_ret_t ret;
+  rcutils_ret_t rcutils_ret;
+  uint8_t uuid[UUID_SIZE];
+  char * uuid_str = NULL;
+  size_t size;
+  char * feedback_filter = NULL;
+  char * feedback_filter_update = NULL;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+
+  feedback_filter = rcutils_strdup("", allocator);
+  if (NULL == feedback_filter) {
+    RCL_SET_ERROR_MSG("failed to allocate memory for feedback filter string");
+    ret = RCL_RET_BAD_ALLOC;
+    goto clean;
+  }
+
+  RCL_RET_FROM_RCUTIL_RET(
+    ret, rcutils_hash_map_get_size(&action_client->impl->goal_uuids, &size));
+  if (RCL_RET_OK != ret) {
+    goto clean;
+  }
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "size: %zu", size);
+  if (0 == size) {
+    // mainly to reset content filtered topic with empty string
+    goto set_cft;
+  }
+
+  RCL_RET_FROM_RCUTIL_RET(
+    ret, rcutils_hash_map_get_next_key_and_data(
+      &action_client->impl->goal_uuids, NULL, uuid, &uuid_str));
+  if (RCL_RET_OK != ret) {
+    goto clean;
+  }
+
+  feedback_filter_update =
+    rcutils_format_string(allocator, "goal_id.uuid = &hex(%s)", uuid_str);
```

Comment:
just checking, i wasn't clear before but this `hex` is not in the spec for interoperability? [Annex B - Syntax for Queries and Filters](https://www.omg.org/spec/DDS/1.4/PDF)?

Reply:
Yes, it's not the official spec.